src/default_config.ts

```diff
@@ -0,0 +1,1212 @@
+/**
```

Comment:
Why not integration `default_config.ts` inside `config.ts`?
Splitting it into two files annoys me a little (though I could be convinced). Even more considering that an application might want to read it to know what those config properties are about. Here having it directly in `src/config.ts` seems more natural.